unit_testing/test_tidegauge_methods.py

```diff
@@ -0,0 +1,223 @@
+            stats = lowestoft.basic_stats("sea_level", "interp_ssh")
+            lowestoft.basic_stats("sea_level", "interp_ssh", create_new_object=False)
+
+            # TEST: Check new object resembles internal object
+            check1 = all(stats.dataset.error == lowestoft.dataset.error)
+            # TEST: Check lengths and values
+            check2 = stats.dataset.absolute_error.shape[0] == lowestoft.dataset.sea_level.shape[0]
+            self.assertTrue(check1, "check1")
+            self.assertTrue(check2, "check2")
+
+    def test_tidegauge_resample_and_apply_doodsonx0(self):
+
+        with self.subTest("Resample to 1H"):
+            date0 = datetime.datetime(2007, 1, 10)
+            date1 = datetime.datetime(2007, 1, 12)
+            lowestoft = coast.Tidegauge(files.fn_tidegauge, date_start=date0, date_end=date1)
+            lowestoft.resample_mean("sea_level", "1H")
+            td0 = lowestoft.dataset.time_1H[1] - lowestoft.dataset.time_1H[0]
+            check1 = td0.values.astype("timedelta64[h]") == np.timedelta64(1, "h")
+            # TEST: Check length
+            check2 = np.ceil(lowestoft.dataset.time.shape[0] / 4) == lowestoft.dataset.time_1H.shape[0]
+            self.assertTrue(check1, "check1")
+            self.assertTrue(check2, "check2")
+
+        with self.subTest("Apply Doodson x0 filter"):
+            lowestoft.apply_doodson_x0_filter("sea_level_1H")
+
+            # TEST: Check new times are same length as variable
+            check1 = lowestoft.dataset.time_1H.shape == lowestoft.dataset.sea_level_1H_dx0.shape
+            # TEST: Check there are number values in output
+            check2 = False in np.isnan(lowestoft.dataset.sea_level_1H_dx0)
+
+            self.assertTrue(check1, "check1")
+            self.assertTrue(check2, "check2")
+
+        with self.subTest("Plot time series"):
+            f, a = lowestoft.plot_timeseries(["sea_level", "sea_level_1H", "sea_level_1H_dx0"])
+            f.savefig(files.dn_fig + "tidegauge_timeseries.png")
+            plt.close("all")
+
+    def test_load_multiple_tidegauge(self):
+
+        with self.subTest("Load multiple gauge"):
+            date0 = datetime.datetime(2007, 1, 10)
+            date1 = datetime.datetime(2007, 1, 12)
+            lowestoft = coast.Tidegauge(files.fn_tidegauge, date_start=date0, date_end=date1)
+            tidegauge_list = coast.Tidegauge.create_multiple_tidegauge(files.fn_multiple_tidegauge, date0, date1)
+
+            # TEST: Check length of list
+            check1 = len(tidegauge_list) == 2
+            # TEST: Check lowestoft matches
+            check2 = all(tidegauge_list[1].dataset == lowestoft.dataset)
+
+            self.assertTrue(check1, "check1")
+            self.assertTrue(check2, "check2")
+
+        with self.subTest("Plot multiple gauge"):
+            f, a = coast.Tidegauge.plot_on_map_multiple(tidegauge_list)
+            f.savefig(files.dn_fig + "tidegauge_multiple_map.png")
```

Comment:
The figures has "grid on" or something equivalent, that looks a bit weird as it overlays grid line in the rectangular region spanning the two Lowestoft and Llandudno sites
![tidegauge_multiple_map](https://user-images.githubusercontent.com/22616872/160110992-a4c99730-8df4-4461-ab9f-02acb2b13955.png)
.

Reply:
I don't get this effect actually:
<img width="565" alt="Screenshot 2022-03-25 at 11 38 15" src="https://user-images.githubusercontent.com/32486723/160114232-54b9c355-481d-4e44-929b-f99be99334e1.png">